python/ray/data/dataset.py

```diff
@@ -542,21 +542,27 @@ def process_next_batch(batch: Block) -> Iterator[Block]:
                 batch = BlockAccessor.for_block(batch).to_batch_format(batch_format)
                 # Apply UDF.
                 batch = batch_fn(batch, *fn_args, **fn_kwargs)
+                if isinstance(batch, dict) and not all(
+                    isinstance(col, np.ndarray) for col in batch.values()
+                ):
```

Comment:
how about using https://note.nkmk.me/en/python-numpy-set-printoptions-threshold/ to configure printing outputs to print truncated tensor ? Pytorch by default has it enabled https://stackoverflow.com/questions/52673610/printing-all-the-contents-of-a-tensor

Also for this conditional I think it will be much more helpful to indicate which column is failing with representative output rather than catching all. p13n track had the same issue with multi-column pandas that figuring out which column went wrong took forever

Reply:
Tried setting `numpy.printoptions`.`threshold` doesn't help because we're already meeting the threshold. Also tried `edgeitems=1`, but the output still felt spammy.

In any case, I've updated the error message to be slightly more descriptive.

Lemme know what you think.

Reply:
i see, thanks for trying ! This is good ux improvement of status quo.